### ISSUE TITLE
CakePHP 3.0 plugins into vendor folder

### ISSUE THREAD

**Issue by Laykou** [composer/installers#217]
According to the new logic introduced in cakephp/cakephp#5641 by @markstory the CakePHP plugins `"type": "cakephp-plugin"` should be installed into the `vendor` folder. The CakePHP has already a [postAutoloadDump](https://github.com/cakephp/plugin-installer/blob/master/src/Installer/PluginInstaller.php#L114) which will link the plugins from the vendor folder into the cake app.

What I see as the problem is that this applies only for the new CakePHP 3.0 version, however there is no difference in the `"type": "cakephp-plugin"` composer attribute for the 2.x versions.

Possibly all the creators of CakePHP 3.x plugins should be instructed not to require this _composer/installers_ plugin in their `composer.json`.

Refs cakephp/cakephp#5918

---

I experienced the issue with the boostrap3 helper plugin where I also created the pull request Holt59/cakephp3-bootstrap3-helpers#10, but globally the issue cannot be fixed there.

**Comment by Laykou** [comment]
The best solution would be to skip the CakePHP processing if the file `vendor/cakephp-plugins.php` exists. This is created only in 3.x to manage the plugin integration into the app.

**Comment by markstory** [comment]
Yes, 3.0 plugins should not require composer/installers. I can update the documentation for CakePHP. We could emit a warning if composer/installers is used with 3.0 plugins.

**Comment by AD7six** [comment]
@markstory would it not be sufficient to change [the check for 3.x](https://github.com/composer/installers/blob/master/src/Composer/Installers/CakePHPInstaller.php#L36-L38) to put them in the vendors folder?

No plugin needs to require this repo - but many still do.

**Comment by Laykou** [comment]
Correct me if I'm wrong, but I would do
```PHP
if ($this->matchesCakeVersion('>=', '3.0.0')) {
    $this->locations['plugin'] = 'vendor/cakephp/cakephp/';
}
```

Not sure if something can go wrong if it's hardcoded this way but I couldn't figure out how force the default behavior with the `vendor` folder, because if the `getLocations()` function returns something which equals to false, an [exception will be thrown](https://github.com/composer/installers/blob/master/src/Composer/Installers/BaseInstaller.php#L63-L65).

**Comment by a-alekseyev** [comment]
Hello, I got this message on tests
1) Composer\Installers\Test\CakePHPInstallerTest::testGetLocations
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'vendor/{$vendor}/{$name}/'
+'/vendor/{$vendor}/{$name}/'

/Users/anton/Documents/Dev/MageCore/installers/tests/Composer/Installers/Test/CakePHPInstallerTest.php:103
                                         
FAILURES!                                
Tests: 225, Assertions: 256, Failures: 1.

**Comment by sbuzonas** [comment]
This is addressed by #219

**Comment by ravage84** [comment]
Yes, this is fixed. All of the three "cakephp-plugin" type packages in my app are installed into the vendor folder.

This can be closed.
